libgadget/treewalk.c

```diff
@@ -21,6 +18,9 @@ static int *Exportnodecount;
 static int *Exportindex;
 static int *Send_offset, *Send_count, *Recv_count, *Recv_offset;
 
+/*!< Memory factor to leave for (N imported particles) > (N exported particles). */
+static double ImportBufferBoost;
```

Comment:
Just looking at this. I am thinking maybe in the future we can have a 
extern struct TreeWalkConfig {

} TreeWalkConfig

In treewalk.h

and 

struct TreeWalkConfig TreeWalkConfig;

in treewalk.c

Then instead of passing in unamed parameters with init_treewalk, we can explicitly set the parameters at in params.c before calling init_treewalk_module().

Reply:
Also a good idea